airbyte-integrations/connectors/source-netsuite/source_netsuite/source.py

```diff
@@ -0,0 +1,135 @@
+            url = base_url + RECORD_PATH + "contact"
+            try:
+                response = session.get(url=url, params={"limit": 1})
+                response.raise_for_status()
+                return True, None
+            except requests.exceptions.HTTPError as e:
+                return False, e
+
+    def get_schemas(self, object_names: Union[List[str], str], session: requests.Session, metadata_url: str) -> Mapping[str, Any]:
+        # fetch schemas
+        if isinstance(object_names, list):
+            return {object_name: session.get(metadata_url + object_name, headers=SCHEMA_HEADERS).json() for object_name in object_names}
+        elif isinstance(object_names, str):
+            return {object_names: session.get(metadata_url + object_names, headers=SCHEMA_HEADERS).json()}
+
+    def generate_stream(
+        self,
+        session: requests.Session,
+        metadata_url: str,
+        schemas: dict,
+        object_name: str,
+        auth: OAuth1,
+        base_url: str,
+        start_datetime: str,
+        window_in_days: int,
+    ) -> Union[NetsuiteStream, IncrementalNetsuiteStream, CustomIncrementalNetsuiteStream]:
+        # try to build the stream instance dynamicaly
+        try:
+            schema = schemas[object_name]
+            schema_props = schema["properties"]
+            if schema_props:
+                if "lastModifiedDate" in schema_props.keys():
+                    # if stream has `lastModifiedDate` as cursor - it's incremental
+                    return IncrementalNetsuiteStream(auth, object_name, base_url, start_datetime, window_in_days)
+                elif "lastmodified" in schema_props.keys():
+                    # if stream has `lastmodified` as cursor - it's custom incrermental
+                    return CustomIncrementalNetsuiteStream(auth, object_name, base_url, start_datetime, window_in_days)
+                else:
+                    # all other streams are full_refresh
+                    return NetsuiteStream(auth, object_name, base_url, start_datetime, window_in_days)
+        except KeyError:
+            print(f"Object `{object_name}` schema has missing `properties` key. Retry...")
+            # somethimes object metadata returns data with missing `properties` key,
+            # we should try to fetch metadata again to that object
+            self.get_schemas(object_name, session, metadata_url)
+            self.generate_stream(session, metadata_url, schemas, object_name, auth, base_url, start_datetime, window_in_days)
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        base_url = self.base_url(config)
+        metadata_url = base_url + META_PATH
+        auth = self.auth(config)
+        session = self.get_session(auth)
+        object_names = config.get("object_types", [])
+        start_datetime = config["start_datetime"]
+        window_in_days = config["window_in_days"]
```

Comment:
this is not a required field

Reply:
it's required by stream instance inside __init__